File: ros_tutorial_cpp/src/listener.cpp
/*
  Filename: listener.cpp
  Author: Ryan Shim
*/

#include <ros/ros.h>
#include <std_msgs/Int8.h>

void countCallback(const std_msgs::Int8::ConstPtr& msg)
{
  ROS_INFO("count: %d", msg->data);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");
  ros::NodeHandle nh;
  ros::Subscriber count_sub = nh.subscribe("count", 10, countCallback);
  ros::spin();
  return 0;
}

File: ros_tutorial_py/scripts/listener.py
#!/usr/bin/env python
'''
  Filename: listener.py
  Author: Ryan Shim
'''

import rospy
from std_msgs.msg import Int8

def count_callback(msg):
    rospy.loginfo("count: %d", msg.data)
    
def listener():
    rospy.init_node('listener', anonymous=True)
    rospy.Subscriber('count', Int8, count_callback)
    rospy.spin()

if __name__ == '__main__':
    listener()
